elements/lisk-api-client/src/ws_channel.ts

```diff
@@ -108,15 +111,33 @@ export class WSChannel {
 			return Promise.resolve();
 		}
 
-		return new Promise<void>(resolve => {
-			this._ws?.on('close', () => {
-				this._ws?.terminate();
-				this.isAlive = false;
-				this._ws = undefined;
-				resolve();
-			});
-			this._ws?.close();
+		if (this._ws.readyState === WebSocket.CLOSED) {
+			this.isAlive = false;
+			this._ws = undefined;
+			return Promise.resolve();
```

Comment:
```
return;
```
should be ok here

Reply:
Yes it should be but ESLint is complaining for consistent promise return.

Reply:
if you consistently use `return`, it should be ok, also the bottom return is not necessary